retrieval/scrape.go

```diff
@@ -807,8 +879,11 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	ref, ok := sl.refCache[s]
+	// Suffix s with the invalid \xff unicode rune to avoid collisions
+	// with scraped metrics.
+	s2 := s + "\xff"
```

Comment:
@brian-brazil I already mean only series that have identical label sets, an `up{foo="bar"}` coming from the target would of course not collide with anything. But what if the target just exposes `up`, and then stops exposing it? It would lead to a stale marker being written, and the synthetic metric then not being written for that iteration (due to same timestamp)?